tests/schema/tests.py

```diff
@@ -3175,3 +3175,68 @@ class Meta:
         with connection.schema_editor(atomic=True) as editor:
             editor.alter_db_table(Foo, Foo._meta.db_table, 'renamed_table')
         Foo._meta.db_table = 'renamed_table'
+
+    @isolate_apps('schema')
+    @unittest.skipIf(
+        connection.vendor == 'postgresql' and not connection.features.is_postgresql_12,
+        'PG < 12 has no non-deterministic collations'
+    )
+    def test_collations(self):
```

Comment:
While having a test for cross-backend CI collation support is great I think having one that runs on all supported database version and only ensures a particular collation is added to the column would also be necessary.

This will likely involve augmenting `introspection` features of database backends to be able to retrieve the collation of a column.

Reply:
I've added the introspection code and some tests using it. I haven't touched Oracle yet, I need to get it installed locally, will take a look at that when I get time.

I imagine the original test isn't really necessary now, but I'll leave it in for the moment.

Reply:
@charettes, I've made a first pass at this, and it all seems to be working across the board. Would you be able to take a look?